src/input_common/sdl/sdl_impl.cpp

```diff
@@ -71,6 +72,38 @@ class SDLJoystick {
         return state.axes.at(axis) / 32767.0f;
     }
 
+    bool RumblePlay(f32 amp_low, f32 amp_high, int time) {
+        const u16 raw_amp_low = static_cast<u16>(amp_low * 0xFFFF);
+        const u16 raw_amp_high = static_cast<u16>(amp_high * 0xFFFF);
+        // Lower drastically the number of state changes
+        if (raw_amp_low >> 11 == last_state_rumble_low >> 11 &&
+            raw_amp_high >> 11 == last_state_rumble_high >> 11) {
+            if (!(raw_amp_low + raw_amp_high == 0 &&
```

Comment:
Inverting the not allows this to be written without it needing to be present in the condition itself:

```suggestion
            if (raw_amp_low + raw_amp_high != 0 ||
                last_state_rumble_low + last_state_rumble_high == 0) {
```